### PR TITLE
Fixed the issue of Home page with white background when navigating by clicking on the SODA Logo or by changing the URL from the address bar. Also fixes active menu item display.

### LINKED ISSUE #656: clicking on soda logo at home page redirect to home with white background
**Issue/Feature Description:**
log in and go to a profile page. then click on the soda logo. you will be redirected to the home page. but it will come in white background. which ideally should be black.

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
this issue causes discrepancy for the user. also, it distorts the symmetry of the soda dashboard.

**How to reproduce, in case of a bug:**
login -> profile -> click on soda logo
**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**
#SODACODE2022 label

### LINKED ISSUE #658: Search from address bar does not change the active item in sidebar menu
**Issue/Feature Description:**
Search from the address bar does not change the active item in the sidebar menu.

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
when the user changes the URL in the address bar soda dashboard takes the user to the desired location. but the active menu in the sidebar is not changed. this causes confusion for the user and decreases the UX.

**How to reproduce, in case of a bug:**
login -> go to buckets -> type /profile in address bar.
Screenshot (28)
Screenshot (29)
SODA Dashboard

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by ravisaxena23**
**What type of PR is this?**
 /kind bug fix
 /kind cleanup
 /kind enhancement

**What this PR does / why we need it**:
This PR solves the UX issues and enhances the overall dashboard functionality. In this PR we have solved the issue relating to the wrong active item shown when the URL is changed from the address bar. Also, this PR solves the issue in which the home page is coming with white background. In this PR we have changed the condition for the router outlet and active menu item.

**Which issue(s) this PR fixes**:
Fixes #658
Fixes #656

**Test Report Added?**:
 /kind TESTED

**Test Report**:
Screenshot (29)
Screenshot (30)
Screenshot (31)
Screenshot (32)

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ravisaxena23** [comment]
fixes #658

File: src/app/app.component.html [hunk 1]
```diff
@@ -57,26 +57,26 @@ <h1>Welcome {{ username }}</h1>
...
                     <ul class="menuGroup-list">
                         <li class="menuGroup-child" *ngFor="let child of menuItem.children">
-                            <a *ngIf="!child.is_external_link" href="#" [routerLink]="[child.routerLink]" [ngClass]="{'active-item':child==activeItem}" (click)="menuItemClick($event, child)">
+                            <a *ngIf="!child.is_external_link" href="#" [routerLink]="[child.routerLink]" [ngClass]="{'active-item':child.routerLink==activeItem.routerLink}" (click)="menuItemClick($event, child)">
                                 <h1>
                                     {{child.title}}
                                 </h1>
                             </a>
-                            <a *ngIf="child.is_external_link" [href]="[child.link]" target="_blank" [ngClass]="{'active-item':child==activeItem}">
+                            <a *ngIf="child.is_external_link" [href]="[child.link]" target="_blank" [ngClass]="{'active-item':child.routerLink==activeItem.routerLink}">
```
Comment (anvithks) [inline src/app/app.component.html:79]: Is this comparison needed? The external link opens in another tab. Will the menu item get highlighted in this case?

**Review by anvithks — APPROVED** [review]
Overall LGTM. 
Please check the other comments on external link.

**Comment by anvithks** [comment]
#SODA-PR-ACCEPTED

**Review by joseph-v — APPROVED** [review]
LGTM

**Comment by anvithks** [comment]
@ravisaxena23 Could you please resolve the conflict.

**Pull request merged**